### PR TITLE
changed Tie Advanced, added Wing Leader and attempted to write the Wing Leader unit test.

### PR DESCRIPTION
**Pull request by redwulfe**
I noticed that the template I was using for Wing leader called for two constants that it did not use and removed those. Then I copied the Tie Advance to implement the wing leader. I then used the Tie Advance unit test to create a unit test for Wing leader.

### CODE REVIEW

**Review by adbronico — CHANGES_REQUESTED** [review]

File: test/server/cards/01_SOR/WingLeader.spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                    },
+                    player2: {
+                        groundArena: ['viper-probe-droid']
+                    }
+                });
+            });
+
+            it('can give two experience to a unit', function () {
+                this.player1.clickCard(this.wingLeader);
+                expect(this.player1).toBeAbleToSelectExactly([this.fleetLieutenant, this.monMothmaVoiceOfTheRebellion]);
```
[the remark this answers is not in the document]
Comment (redwulfe) [reply test/server/cards/01_SOR/WingLeader.spec.js]: desided to switch from mon mothma to rebel pathfinder as I could not get mothma to pass correctly.

File: server/game/cards/01_SOR/TieAdvanced.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default class TieAdvanced extends NonLeaderUnitCard {
 
     public override setupCardAbilities() {
         this.addWhenPlayedAbility({
-            title: 'Give 2 experience tokens to a friendly Imperial unit',
+            title: 'Give 2 Experience tokens to a friendly IMPERIAL unit',
```
Comment (adbronico) [inline server/game/cards/01_SOR/TieAdvanced.ts:15]: I would revert IMPERIAL to Imperial here

**Review by adbronico — CHANGES_REQUESTED** [review]
One small change, otherwise looks good

File: server/game/cards/01_SOR/WingLeader.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    protected override getImplementationId() {
+        return {
+            id: '3443737404',
+            internalName: 'wing-leader',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'Give 2 Experience tokens to another friendly REBEL unit',
```
Comment (adbronico) [inline server/game/cards/01_SOR/WingLeader.ts:15]: Change REBEL to Rebel

File: server/game/cards/01_SOR/TieAdvanced.ts (hunk 2 above)
Comment (redwulfe) [reply server/game/cards/01_SOR/TieAdvanced.ts:15]: will do

**Review by adbronico — APPROVED** [review]
This is where the fun begins

File: test/server/cards/01_SOR/WingLeader.spec.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                    }
+                });
+            });
+
+            it('can give two experience to a unit', function () {
+                this.player1.clickCard(this.wingLeader);
+                expect(this.player1).toBeAbleToSelectExactly([this.fleetLieutenant, this.rebelPathfinder]);
+                this.player1.clickCard(this.fleetLieutenant);
+
+                expect(this.fleetLieutenant.upgrades.map((upgrade) => upgrade.internalName)).toEqual(['experience', 'experience']);
```
Comment (AMMayberry1) [inline test/server/cards/01_SOR/WingLeader.spec.js:22]: I actually just merged in a new test method that makes this a little easier. Do you mind updating this to be:

`expect(this.fleetLieutenant).toHaveExactUpgradeNames(['experience', 'experience']);`

I already updated the TIE Advanced test, so this way everything will be consistent and other people will know to use this method in the future if they look at this code as an example. Ditto for the expectation on line 30.

Comment (redwulfe) [reply test/server/cards/01_SOR/WingLeader.spec.js:22]: will do.

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**